front/lib/api/assistant/configuration.ts

```diff
@@ -740,6 +747,12 @@ export async function createAgentConfiguration(
   let version = 0;
   let listStatusOverride: AgentUserListStatus | null = null;
 
+  const groupIds = (
+    await DataSourceViewResource.fetchByIds(auth, dataSourceViewIds)
+  )
+    .map((view) => view.acl().aclEntries.map((entry) => entry.groupId))
```

Comment:
```suggestion
    .flatMap((view) => view.acl().aclEntries.map((entry) => entry.groupId))
```

Reply:
is `.flatMap` really that much better than .map().flat()?

Reply:
If it's exist, I'd say let's use it 🙃.